pkg/ebpf/netlink/decoding.go

```diff
@@ -0,0 +1,148 @@
+package netlink
+
+import (
+	"encoding/binary"
+	"net"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	ct "github.com/florianl/go-conntrack"
+)
+
+const (
+	ctaUnspec = iota
+	ctaTupleOrig
+	ctaTupleReply
+)
+
+const (
+	ctaTupleIP    = 1
+	ctaTupleProto = 2
+	ctaTupleZone  = 3
+)
+
+const (
+	ctaIPv4Src = 1
+	ctaIPv4Dst = 2
+	ctaIPv6Src = 3
+	ctaIPv6Dst = 4
+)
+
+const (
+	ctaProtoNum     = 1
+	ctaProtoSrcPort = 2
+	ctaProtoDstPort = 3
+)
+
+var scanner = NewAttributeScanner()
+
+// TODO: Replace the intermediate ct.Con object by the same format we use in the cache
+func DecodeEvent(e Event) []ct.Con {
+	msgs := e.Messages()
+	conns := make([]ct.Con, 0, len(msgs))
```

Comment:
Does it make sense to use `ForEach/Iterate` pattern here and pass a callback method to accept each parsed connection rather than allocating a slice for every event?

Reply:
It probably does. The only downside I can think of would be mutex contention, because we could register the `[]ct.Con` in the `state` cache. but I suspect it's not highly content mutex, since we mostly acquire it during the checks. anyway, we can validate it in a blocking profile.